### PR TITLE
Fixed root.sh to properly find for libxml2

### PR DESCRIPTION
**Pull request by jan-busa**
When building ROOT v6-24-06 with libxml2/v2.9.3 (--no-system build), proper variable for LIBXML2 libdir and include dir has to be set in ROOT configuration.

Signed-off-by: Jan Busa Jr <busa@jinr.ru>

### CODE REVIEW

File: root.sh [hunk 1]
```diff
@@ -127,6 +127,8 @@ cmake $SOURCEDIR
       ${OPENSSL_ROOT:+-DOPENSSL_ROOT=$OPENSSL_ROOT}                                    \
       ${OPENSSL_ROOT:+-DOPENSSL_INCLUDE_DIR=$OPENSSL_ROOT/include}                     \
       ${LIBXML2_ROOT:+-DLIBXML2_ROOT=$LIBXML2_ROOT}                                    \
+      ${LIBXML2_ROOT:+-DPC_LIBXML_INCLUDEDIR=$LIBXML2_ROOT/include}                    \
+      ${LIBXML2_ROOT:+-DPC_LIBXML_LIBDIR=$LIBXML2_ROOT/lib}                            \
```
Comment (ktf) [inline root.sh:131]: I am surprised one needs to override the package config hooks. Which distribution are you using?

**Review by ktf — COMMENTED** [review]
Which distribution are you using?

File: root.sh (hunk 1 above)
Comment (jan-busa) [reply root.sh:131]: I'm doing build inside CentOS-7 and using alibuild --no-system build. This builds CMake 3.19.2 and inside Modules directory is libXML2 expecting the variable PC_LIBXML_LIBDIR (see slc7_x86-64/CMake/v3.19.2-X/share/cmake-3.19/Modules/FindLibXml2.cmake). LibXML2 is not from system but was built using alibuild.

**Comment by ktf** [comment]
@TimoWilken could you warm up the cache with this PR? I think it's ready to go.

**Comment by jan-busa** [comment]
I would like to add one more thing to the root.sh, can you wait with merge or should I make another pull request? I would like to add possibility to build root with fftw3 support (allowing both external FFTW3 library or enabling builtin-fftw3).

**Comment by ktf** [comment]
Ok, let us know when you are ready.

**Pull request closed without merging**